web/public/locales/en.json

```diff
@@ -49,6 +49,13 @@
         "noParserInfo": "We do not yet have any information about this area.<br/> Want to help fix this? You can contribute by <a href=\"%s\" target=\"_blank\">adding area data</a>.",
         "now": "Now"
     },
+    "time-controller": {
+        "title": "Display data from the last",
+        "day": "Day",
+        "month": "Month",
+        "year": "Year",
+        "5year": "5 years"
```

Comment:
Consider using `Intl.DisplayNames()` and Intl.NumberFormat() for these instead then we don't need to include them in the translation files.

Code needed:
```js
new Intl.DisplayNames(document.documentElement.lang, {type: 'dateTimeField'}).of('day')
```
and:
```js
new Intl.NumberFormat(document.documentElement.lang,  { style: 'unit', unit: 'year', unitDisplay: 'long' }).format(5);
```
[MDN | Intl.DisplayNames() constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/DisplayNames#using_type_datetimefield)
[MDN | Intl.NumberFormat() constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat)

Reply:
I agree, that would be a nice approach :)

Reply:
@Kongkille that is why I used `document.documentElement.lang` we already set the document language with i18n in `i18n.js`.
https://github.com/electricitymap/electricitymap-contrib/blob/346d621810809ce2f7359212fb092ebeedb2a052/web/src/helpers/i18n.js#L80-L81

I have used `document.documentElement.lang` in other places but we could change those to match, either way it would be good to decide how we want to do it so the codebase is consistent.

Reply:
Well if I understand correctly, I don't think we want to rely on a document attribute in all places.

In my PR i remove this line so we no longer use this hook. This means that (because TimeAxis is a memo component) we don't actually re-render the component onLanguageChange.
https://github.com/electricitymap/electricitymap-contrib/blob/346d621810809ce2f7359212fb092ebeedb2a052/web/src/components/graph/timeaxis.jsx#L31

Essentially this means that language specific changes are not going through unless the component is guaranteed to rerender. Right now it works on master because they all somehow use the hook, but it can lead to some odd behavior.

As for the Memo I think you're right, I am not sure what the cost of it is but unless it's really expensive we should probably avoid it, if not for readability

Reply:
You can see here that the TimeAxis actually don't re-render on this PR


https://user-images.githubusercontent.com/45588799/167462682-1837d652-fa3d-4ea9-9920-4d515dec03b3.mov

So I think that we should use the "useTranslation" hook to get the language as well (or create a new hook specifically to get the languageCode perhaps)

Reply:
As we aren't passing this global object to the function/component but rather using it directly don't this make memo unnecessary? (I haven't used react much so please correct me if I'm wrong).

Originally I figured it would be best to use a global object over function calls/imports as they would have lower performance overhead but it probably don't matter much when we only do a few calls each time. 

However if we do need memo I think it does make sense to create a languageCode hook for clarity and consistency.

Reply:
I think certainly some of our Memo components/functions are not necessary but right now we have them in the codebase, so my point is moreso that we risk the components not rerendering.

It seems to be that the most 'react' approach would be to have a languageCode hook so we absolutely guarantee that the component rerenders when language changes.

In the future we could perhaps look into some of these memos to determine how useful they are :)